DependencyInjection/Configuration.php

```diff
@@ -40,25 +43,92 @@ public function getConfigTreeBuilder() : TreeBuilder
         $organizeMigrationModes = $this->getOrganizeMigrationsModes();
 
         $rootNode
+            ->fixXmlConfig('migration', 'migrations')
+            ->fixXmlConfig('migrations_path', 'migrations_paths')
             ->children()
-                ->scalarNode('dir_name')->defaultValue('%kernel.root_dir%/DoctrineMigrations')->cannotBeEmpty()->end()
-                ->scalarNode('namespace')->defaultValue('Application\Migrations')->cannotBeEmpty()->end()
-                ->scalarNode('table_name')->defaultValue('migration_versions')->cannotBeEmpty()->end()
-                ->scalarNode('column_name')->defaultValue('version')->end()
-                ->scalarNode('column_length')->defaultValue(14)->end()
-                ->scalarNode('executed_at_column_name')->defaultValue('executed_at')->end()
-                ->scalarNode('all_or_nothing')->defaultValue(false)->end()
                 ->scalarNode('name')->defaultValue('Application Migrations')->end()
-                ->scalarNode('custom_template')->defaultValue(null)->end()
-                ->scalarNode('organize_migrations')->defaultValue(false)
+
+                ->arrayNode('migrations_paths')
+
+                    ->info('A list of pairs namespace/path where to look for migrations.')
+                    ->requiresAtLeastOneElement()
+                    ->useAttributeAsKey('namespace')
+                    ->defaultValue(['%kernel.project_dir%/src/Migrations' => 'App\Migrations'])
+                    ->prototype('scalar')->end()
+                 ->end()
+
+                ->arrayNode('services')
```

Comment:
i'm not 100% sure this is a good idea.... but it will avoid having to do custom stuff for each service

Reply:
Not sure either… I wonder how this is usually done in other bundles

Reply:
Since `DependencyFactory` exposes already the ability to change each of its services, this should not hurt more than it.